public/service-worker.js

```diff
@@ -0,0 +1,40 @@
+// https://deanhume.com/create-a-really-really-simple-offline-page-using-service-workers/
+'use strict';
+
+var cacheVersion = 1;
+var currentCache = {
+  offline: 'govuk-calendar-cache-' + cacheVersion
+};
+const offlineUrls = [
+  '/bank-holidays',
+  '/when-do-the-clocks-change'
+];
+
+this.addEventListener('install', event => {
+  this.skipWaiting();
+  event.waitUntil(
+    caches.open(currentCache.offline).then(function(cache) {
+      return cache.addAll(offlineUrls);
+    })
+  );
+});
+
+this.addEventListener('fetch', event => {
+  // request.mode = navigate isn't supported in all browsers
+  // so include a check for Accept: text/html header.
+  if (event.request.mode === 'navigate' || (event.request.method === 'GET' && event.request.headers.get('accept').includes('text/html'))) {
+    event.respondWith(
+      fetch(event.request.url).catch(error => {
+        // Return the offline page
+        return caches.match(event.request.url);
+      })
+    );
+  } else {
+    // Respond with everything else if we can
```

Comment:
Had to re-read this to understand that this isn't returning pages, but assets.

```suggestion
    // This is an asset - attempt to serve from cache, fall back to network
```

Love this PR! Can't wait to see it merged 🎉